### ISSUE TITLE
Security Error calling PreApplicationStartCode.Start()

### ISSUE THREAD

**Issue by adamvaul** [Offroadcode/SuperchargedReact.Net#3]
Security Error calling PreApplicationStartCode.Start()

Steps to Reproduce:
1. Downloaded zip file.
2.  Without making any changes to the project, running with Debug > Start Without Debugging.

Full stack trace below:

Server Error in '/' Application.

Attempt by security transparent method 'System.Web.Mvc.PreApplicationStartCode.Start()' to access security critical method 'System.Web.WebPages.Razor.PreApplicationStartCode.Start()' failed.

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.MethodAccessException: Attempt by security transparent method 'System.Web.Mvc.PreApplicationStartCode.Start()' to access security critical method 'System.Web.WebPages.Razor.PreApplicationStartCode.Start()' failed.

Source Error: 

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Stack Trace: 

[MethodAccessException: Attempt by security transparent method 'System.Web.Mvc.PreApplicationStartCode.Start()' to access security critical method 'System.Web.WebPages.Razor.PreApplicationStartCode.Start()' failed.]
   System.Web.Mvc.PreApplicationStartCode.Start() +35

[InvalidOperationException: The pre-application start initialization method Start on type System.Web.Mvc.PreApplicationStartCode threw an exception with the following error message: Attempt by security transparent method 'System.Web.Mvc.PreApplicationStartCode.Start()' to access security critical method 'System.Web.WebPages.Razor.PreApplicationStartCode.Start()' failed..]
   System.Web.Compilation.BuildManager.InvokePreStartInitMethodsCore(ICollection`1 methods, Func`1 setHostingEnvironmentCultures) +556
   System.Web.Compilation.BuildManager.InvokePreStartInitMethods(ICollection`1 methods) +132
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +102
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +153
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +516

[HttpException (0x80004005): The pre-application start initialization method Start on type System.Web.Mvc.PreApplicationStartCode threw an exception with the following error message: Attempt by security transparent method 'System.Web.Mvc.PreApplicationStartCode.Start()' to access security critical method 'System.Web.WebPages.Razor.PreApplicationStartCode.Start()' failed..]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +9885060
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +101
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.0.30319.34009

**Comment by PeteDuncanson** [comment]
Wow, great bug report. I'll see if I can recreate at my end.

**Comment by adamvaul** [comment]
Hey Pete,

I just want to say I am a big supporter of all you are doing.  I am looking
to use React at my current company but I can't endorse it until I really
get all of the kinks worked out like how to do routing properly.

I was excited to see your other Githhub repos for GraphQL and things like
that.  Really good stuff.

Yeah, please see if you can fix those issues (there is another one I
submitted for the https://github.com/Offroadcode/Umbraco-React-Starter-kit
project as well)

I have been following this thread and posted the issues there too.
https://github.com/reactjs/React.NET/issues/64

**Comment by steroberts89** [comment]
Hey,

what was happening was Visual studio was trying to run the project Orc.SuperchargedReact.Web as an MVC site, but this project was being used more as a class library. I've made the change so Visual studio won't think it's a website.

Thanks
Stephen

**Comment by adamvaul** [comment]
Thanks Pete,

I got the latest and it when I tried to build it complained about different versions of Newtonsoft.Json being used in different projects.  The Orc.SuperchargedReact.Runner was using 4.0 or something and the Orc.SuperchargedReact.Core was using 6.0.  Anyway, so I just removed Newtonsoft.Json and updated all three of them to the latest version 7.0.0.0 and it builds successfully.

Now what am I supposed to do with it?
